crates/wasi-common/src/ctx.rs

```diff
@@ -407,18 +424,51 @@ impl WasiCtx {
     /// The `Entry` will automatically get another free raw WASI `fd` assigned. Note that
     /// the two subsequent free raw WASI `fd`s do not have to be stored contiguously.
     pub(crate) fn insert_entry(&self, entry: Entry) -> Result<Fd, Error> {
-        self.entries.borrow_mut().insert(entry).ok_or(Error::Mfile)
+        self.inner
+            .entries
+            .borrow_mut()
+            .insert(entry)
+            .ok_or(Error::Mfile)
     }
 
     /// Insert the specified `Entry` with the specified raw WASI `fd` key into the `WasiCtx`
     /// object.
     pub(crate) fn insert_entry_at(&self, fd: Fd, entry: Rc<Entry>) {
-        self.entries.borrow_mut().insert_at(&fd, entry)
+        self.inner.entries.borrow_mut().insert_at(&fd, entry)
     }
 
     /// Remove `Entry` corresponding to the specified raw WASI `fd` from the `WasiCtx` object.
     pub(crate) fn remove_entry(&self, fd: Fd) -> Result<Rc<Entry>, Error> {
-        self.entries.borrow_mut().remove(fd).ok_or(Error::Badf)
+        self.inner
+            .entries
+            .borrow_mut()
+            .remove(fd)
+            .ok_or(Error::Badf)
+    }
+
+    /// Insert the specified `Handle` into the entry table.
+    pub fn insert_handle<T: Handle + 'static>(&mut self, handle: T) -> Result<Fd, Error> {
```

Comment:
Should this be `&self` instead of `&mut self`?

Reply:
Fixed; thanks!